[13.0][MIG] mig_crm_claim: Migration to 13.0

manuelcalerosolis opened the pull request:
This is the migration mig_crm_claim to 13.0 version

@Tecnativa TT22081

cubells committed:
[MIG] crm_claim: Migration to 10.0

Commit:
[10.0][ENH] crm_claim (#178)

oihane committed:
[MIG] crm_claim: Migration to 11.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

oihane committed:
[11.0][IMP] crm_claim: add activities to claims (#214)

oca-travis committed:
[UPD] Update crm_claim.pot

oihane committed:
[IMP] crm_claim: autogenerated readme

oca-travis committed:
[UPD] Update crm_claim.pot

cristinamartinrod committed:
[MIG] crm_phonecall_summary_predefined: Migration to 11.0

oca-travis committed:
[UPD] Update crm_claim.pot

cristinamartinrod committed:
[MIG] crm_claim: Migration to v12

eicher31 committed:
Fix default date

elrodo1 committed:
[FIX] crm_claim: Lint error was fixed

max3903 committed:
Update crm_claim_report.py

pedrobaeza committed:
[IMP] crm_claim: Multi-company

Include record rule for multi-company.

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 10.5% (15 of 143 strings)

Translation: crm-12.0/crm-12.0-crm_claim
Translate-URL: https://translation.odoo-community.org/projects/crm-12-0/crm-12-0-crm_claim/hr/

malinkienke committed:
Added translation using Weblate (German)

rousseldenis commented:
@manuelcalerosolis Please fix runbot warnings

pedrobaeza commented:
Please squash migration commits and administrative commits

ernestotejeda commented on crm_claim/__manifest__.py:
```diff
@@ -1,11 +1,12 @@
 {
     "name": "Claims Management",
-    "version": "12.0.1.1.0",
+    "version": "13.0.1.1.0",
```
```suggestion
    "version": "13.0.1.0.0",
```

Tardo replied:
Marked as resolved but no changes was made...

ernestotejeda commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "restructuredtext.confPath": ""
+}
```
I think this file shouldn't be here

ernestotejeda commented on crm_claim/report/crm_claim_report_view.xml:
```diff
@@ -70,7 +70,7 @@
         <field name="name">Claims Analysis</field>
         <field name="res_model">crm.claim.report</field>
-        <field name="view_type">form</field>
+        <field name="binding_view_types">form</field>
```
I think you should only add `binding_view_types` in actions that in v12 has `multi = ...`

manuelcalerosolis replied:
Ok thanks!

ernestotejeda reviewed (commented):
Thank you @manuelcalerosolis, Just some suggestions

You can generate README.rst using OCA/maintainer-tools

NL66278 commented on crm_claim/models/crm_claim.py:
```diff
@@ -0,0 +1,182 @@
+            - team_id: if set, stages must belong to this team or
+              be a default case
+        """
+        if domain is None:  # pragma: no cover
```
You will not need the `#pragma` if you put on one line: `domain = domain if domain else []`. That has the added advantage of guarding against exceptions by people passing False or empty strings or other falsey values to domain that are not `[]` or `Nothing`.

manuelcalerosolis replied:
ok many thanks

NL66278 reviewed (changes requested):
:+1: Looks good in general. One nag. And please squash commits as suggested by @pedrobaeza

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (143 of 143 strings)

Translation: crm-12.0/crm-12.0-crm_claim
Translate-URL: https://translation.odoo-community.org/projects/crm-12-0/crm-12-0-crm_claim/es/

Tardo commented on crm_claim/models/crm_claim.py:
```diff
@@ -0,0 +1,181 @@
+            stage_id=self._get_default_stage_id(),
+            name=_("%s (copy)") % self.name,
+        )
+        return super(CrmClaim, self).copy(default)
```
```suggestion
        return super().copy(default)
```

manuelcalerosolis replied:
Done

Tardo commented on crm_claim/models/crm_claim.py:
```diff
@@ -0,0 +1,181 @@
+        if msg.get("priority"):
+            defaults["priority"] = msg.get("priority")
+        defaults.update(custom_values)
+        return super(CrmClaim, self).message_new(msg, custom_values=defaults)
```
```suggestion
        return super().message_new(msg, custom_values=defaults)
```

manuelcalerosolis replied:
Done

Tardo commented on crm_claim/readme/CONTRIBUTORS.rst:
```diff
@@ -0,0 +1,7 @@
+* Odoo S.A.
+* Vicent Cubells <vicent.cubells@tecnativa.com>
```
Move this too

manuelcalerosolis replied:
Done

Tardo commented on crm_claim/readme/CONTRIBUTORS.rst:
```diff
@@ -0,0 +1,7 @@
+* Odoo S.A.
+* Vicent Cubells <vicent.cubells@tecnativa.com>
+* Bhavesh Odedra <bodedra@ursainfosystems.com>
+* Cristina Martin R. <cristina.martin@tecnativa.com>
```
Move this too

manuelcalerosolis replied:
done

Tardo commented on crm_claim/readme/CONFIGURE.rst:
```diff
@@ -0,0 +1,6 @@
+To configure this module, you need to:
```
Please, comment that developer mode is required to view these menus.

manuelcalerosolis replied:
Done

Tardo commented on crm_claim/readme/USAGE.rst:
```diff
@@ -0,0 +1,2 @@
+* Go to new menu **CRM > After Sale > Services > Claims** and create a new
```
```suggestion
* Go to new menu **CRM > After Sale > Claims** and create a new
```

manuelcalerosolis replied:
Done

Tardo reviewed (changes requested):
Thanks, looks good!

Commit:
[IMP] crm_claim: black, isort

Commit:
[MIG] crm_claim: Migration to 13.0

chienandalu commented on crm_claim/report/crm_claim_report.py:
```diff
@@ -1,11 +1,14 @@
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
 
-from psycopg2.extensions import AsIs
+# pylint: disable=sql-injection
```
Is this necessary?

Why not to avoid psycopg2 formatters and use a report style like Odoo does: https://github.com/odoo/odoo/blob/13.0/addons/sale/report/sale_report.py#L61

manuelcalerosolis replied:
@Yajo  and I, we were looking at several ways to solve it, he recommended that I use that directive.

yajo replied:
Indeed much better @chienandalu, thanks for the tip.

carlosdauden commented:
Merged in https://github.com/OCA/crm/pull/316
